zkevm-circuits/src/ecc_circuit.rs

```diff
@@ -0,0 +1,1383 @@
+                QuantumCell::Existing(py_valid),
+                QuantumCell::Existing(p_is_on_curve_or_infinity),
+                QuantumCell::Existing(qx_valid),
+                QuantumCell::Existing(qy_valid),
+                QuantumCell::Existing(q_is_on_curve_or_infinity),
+            ],
+        );
+        let inputs_invalid = ecc_chip
+            .field_chip()
+            .range()
+            .gate()
+            .not(ctx, QuantumCell::Existing(inputs_valid));
+
+        log::trace!("[ECC] EcAdd Inputs Assigned:");
+        log_context_cursor!(ctx);
+
+        // We follow the approach mentioned below to handle many edge cases for the points P, Q and
+        // R so that we can maintain the same fixed and permutation columns and reduce the overall
+        // validation process from the EVM Circuit.
+        //
+        // To check the validity of P + Q == R, we check:
+        // r + P + Q - R == r
+        // where r is a random point on the curve.
+        //
+        // We cover cases such as:
+        // - P == (0, 0) and/or Q == (0, 0)
+        // - P == -Q, i.e. P + Q == R == (0, 0)
+        let res = op.r.unwrap_or(G1Affine::identity());
+        let point_r = self.handle_g1(ctx, ecc_chip, res, powers_of_256);
+        let rx_is_zero = ecc_chip.field_chip.is_zero(ctx, &point_r.ec_point.x);
+        let ry_is_zero = ecc_chip.field_chip.is_zero(ctx, &point_r.ec_point.y);
+
+        let rand_point = ecc_chip.load_random_point::<G1Affine>(ctx);
+        let point_p_is_zero = ecc_chip.field_chip.range().gate().or_and(
+            ctx,
+            QuantumCell::Existing(inputs_invalid),
+            QuantumCell::Existing(px_is_zero),
+            QuantumCell::Existing(py_is_zero),
+        );
+        let point_q_is_zero = ecc_chip.field_chip.range().gate().or_and(
+            ctx,
+            QuantumCell::Existing(inputs_invalid),
+            QuantumCell::Existing(qx_is_zero),
+            QuantumCell::Existing(qy_is_zero),
+        );
+        let point_r_is_zero = ecc_chip.field_chip.range().gate().or_and(
+            ctx,
+            QuantumCell::Existing(inputs_invalid),
+            QuantumCell::Existing(rx_is_zero),
+            QuantumCell::Existing(ry_is_zero),
+        );
+
+        // sum1 = if P == (0, 0) then r else r + P
+        let sum1 = ecc_chip.add_unequal(ctx, &rand_point, &point_p, true);
+        let sum1 = ecc_chip.select(ctx, &rand_point, &sum1, &point_p_is_zero);
+
+        // sum2 = if Q == (0, 0) then sum1 else sum1 + Q
+        let sum2 = ecc_chip.add_unequal(ctx, &sum1, &point_q, true);
+        let sum2 = ecc_chip.select(ctx, &sum1, &sum2, &point_q_is_zero);
+
+        // sum3 = if R == (0, 0) then sum2 else sum2 - R
+        let sum3 = ecc_chip.sub_unequal(ctx, &sum2, &point_r.ec_point, true);
+        let sum3 = ecc_chip.select(ctx, &sum2, &sum3, &point_r_is_zero);
+
+        ecc_chip.assert_equal(ctx, &rand_point, &sum3);
+
+        log::trace!("[ECC] EcAdd Assignmnet END:");
```

Comment:
Nitpick: Assignment

Reply:
Nice catch! fixed in [4d5070f](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1720/commits/4d5070fdf1d7821cab0b4b2caf5f0582fe291ae2)